### PR TITLE
Validate MTA extension descriptor + metadata in mta.yaml

### PR DESCRIPTION
**Pull request by tal-sapan**
Metadata schema is not validated yet (only semantic validations are done).
It will be done in another PR.

### Checklist
- [X] Code compiles correctly
- [X] Relevant tests were added (unit / contract / integration)
- [ ] Relevant logs were added
- [X] Formatting and linting run locally successfully
- [X] All tests passing
- [ ] UA review
- [ ] Design is documented
- [ ] Extended the README / documentation, if necessary
- [ ] Open source is approved

### CODE REVIEW

File: validations/semantic_metadata.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+		if !ok {
+			keyNode := getPropByName(metadataNode, key)
+			issues = append(issues, YamlValidationIssue{Msg: fmt.Sprintf(unknownNameInMetadataMsg, key, mapTypes[mapType].entityKind), Line: keyNode.Line})
+		}
+	}
+	if metadata != nil {
+		for key, value := range m {
+			// If there's no metadata for the key we don't perform this check (since it's overwritable by default)
+			if meta, ok := metadata[key]; ok {
+				if !meta.Optional && !meta.OverWritable && value == nil {
```
Comment (allaVolkov) [inline validations/semantic_metadata.go]: second if condition can be added to the first if

File: validations/ext_validate.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+		s = strings.Replace(s, "\r\n", "\r", -1)
+		yamlContent = []byte(s)
+
+		// Validates MTA content.
+		contentErrIssues, contentWarnIssues := validateExt(yamlContent, projectPath, extFilename,
+			validateSchema, validateSemantic, strict, exclude)
+		errIssues = append(errIssues, contentErrIssues...)
+		warnIssues = append(warnIssues, contentWarnIssues...)
+		if len(errIssues) > 0 {
+			return warnIssues.String(), errors.Errorf(`the "%v" file is not valid: `+"\n%v",
```
[the remark this answers is not in the document]
Comment (tal-sapan) [reply validations/ext_validate.go]: Copied from `MtaYaml` function

File: validations/semantic_metadata.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	resourcesNode := getPropContent(mtaNode, resourcesYamlField)
+	for i, resource := range mta.Resources {
+		issues = append(issues, checkMetadata(resource.Parameters, resource.ParametersMetaData, resourcesNode[i], mapTypeParameters)...)
+		issues = append(issues, checkMetadata(resource.Properties, resource.PropertiesMetaData, resourcesNode[i], mapTypeProperties)...)
+
+		requiresNode := getPropContent(resourcesNode[i], requiresYamlField)
+		issues = append(issues, checkRequiresParamsAndPropertiesMetadata(requiresNode, resource.Requires)...)
+	}
+
+	if strict {
```
[the remark this answers is not in the document]
Comment (tal-sapan) [reply validations/semantic_metadata.go]: I think it's only meant for schema validations, but we should make sure when Natalia is back

File: validations/semantic_metadata.go (hunk 1 above)
Comment (tal-sapan) [reply validations/semantic_metadata.go]: It seemed like too many conditions for 1 if

File: validations/semantic_ext_validate.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package validate
+
+import (
+	"github.com/SAP/cloud-mta/mta"
```
[the remark this answers is not in the document]
Comment (tal-sapan) [reply validations/semantic_ext_validate.go]: Done

**Pull request merged**